neuralop/training/trainer.py

```diff
@@ -197,16 +216,32 @@ def train(
 
                 loss = 0.0
 
-                if self.overrides_loss:
+                if self.overrides_training_loss:
                     loss += self.callbacks.compute_training_loss(
                         out=out, **sample, amp_autocast=self.amp_autocast
                     )
                 else:
-                    if self.amp_autocast:
-                        with amp.autocast(enabled=True):
-                            loss += training_loss(out, **sample)
+                    if self.monitor_individual_train_losses:
```

Comment:
An idea would be to always output a loss dictionary (i.e., automatically use the weighted loss, even for a single loss) to simplify the trainer?